packages/roosterjs-editor-plugins/lib/plugins/TableResize/preprocessTable.ts

```diff
@@ -0,0 +1,98 @@
+import { normalizeRect } from 'roosterjs-editor-dom';
+
+export const enum Dimension {
+    HEIGHT,
+    WIDTH,
+    BOTH,
+}
+
+export const enum ResizeState {
+    None,
+    Horizontal,
+    Vertical,
+    Both, // when resizing the whole table
+}
+
+export function setHTMLElementSizeInPx(element: HTMLElement, dimension: Dimension) {
+    if (!!element) {
+        if (dimension === Dimension.WIDTH || dimension === Dimension.BOTH) {
+            element.removeAttribute('width');
+            element.removeAttribute('height');
+            const unit = element.style?.width?.substr(element.style.width.length - 2);
+            if (!unit || unit !== 'px' /*&& unit !== 'pt'*/) {
+                const rect = normalizeRect(element.getBoundingClientRect());
+                element.style.boxSizing = 'border-box';
+                element.style.width = `${rect.right - rect.left}px`;
+            }
+        }
+
+        if (dimension === Dimension.HEIGHT || dimension === Dimension.BOTH) {
+            const unit2 = element.style?.height?.substr(element.style.height.length - 2);
+            if (!unit2 || unit2 !== 'px' /*&& unit2 !== 'pt' */) {
+                const rect = normalizeRect(element.getBoundingClientRect());
+                element.style.boxSizing = 'border-box';
+                element.style.height = `${rect.bottom - rect.top}px`;
```

Comment:
And set size of one cell may impact the size of other cells. It is also not good for perf.
The best practice is to get all sizes together, then write back together. So 1 read, 1 write, only cause 1 relayout. It is the worst to have read/write/read/write/... loop because each time browser need to relayout.

That is why I think relying on VTable may be good here. You can calculate size for all cells using getBoundingClientRect at once, write the calcualted size to VTable. Then call writeBack() to let VTable help you flush all the calculated sizes, and remove other attributes.

Reply:
It is already one-time calculate during one resizing operation. We have the variables such as isReadyForResizingWidth to ensure all the cell size is already calculated to avoid unnecessary calculation.